### DISCUSSION TITLE
Use Weaviate to improve Proceedings search and linking data

### DISCUSSION THREAD

**Discussion by laura-ham** [SmartDataAnalytics/OpenResearch#122] [Ideas]
This discussion topic is around a potential solution to https://github.com/SmartDataAnalytics/OpenResearch/issues/113. In this issue, a proceedings title parser is proposed. The current implementation looks for exact matches of the words entered in the form and the keywords of the proceedings title. If a similar name is used but the same proceeding was intended to be found, this is not possible with the current implementation. A potential solution is to use [Weaviate](https://github.com/semi-technologies/weaviate) to store the proceedings data. Searches can then be done based on concept or semantically, rather than only with exact matching keywords. 

Additionally, Weaviate could solve data enrichment by making relations between various existing (meta) data sources (e.g. proceedings, conferences, papers, authors, countries, etc).

An example of a successful project with Weaviate and scientific articles from Arxiv can be found [here](https://github.com/semi-technologies/arXiv-demo-dataset). 2.7 GB of scholarly articles with cross references to authors, journals and categories are present in a Weaviate instance. This enables semantic search through these publications and articles, overcoming the exact keyword matching-based search problem. You can try it out yourself with newsarticles in a live demo setting [here](https://console.semi.technology/console/query#weaviate_uri=https://demo.dataset.playground.semi.technology&graphql_query=%7B%0A%20%20Get%7B%0A%20%20%20%20Article(%0A%20%20%20%20%20%20nearText%3A%20%7B%0A%20%20%20%20%20%20%20%20concepts%3A%20%5B%22traveling%20in%20Asia%22%5D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20)%7B%0A%20%20%20%20%20%20title%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) (not Arxiv since it's being updated as I'm typing this). 

First step would be to [create a Weaviate data schema](https://www.semi.technology/developers/weaviate/current/tutorials/how-to-create-a-schema.html) based on the available (meta) data of this project.

Read here on how to get started with a Weaviate instance running locally or in the cloud: https://www.semi.technology/developers/weaviate/current/getting-started/installation.html

I would like to hear your opinions on this idea. I could help with the ideation and implementation if there are questions.

**Comment by WolfgangFahl** [comment]
Hi Laura - welcome to our discussions and thanks for all the effort you and Bob have put into this in recent months. I am positive that Weaviate might be part of our solution in the mid- to long term. Unfortunately, I could not work on my research on that matter in January. Therefore, I might have to postpone things further depending on the priorities of the project. I'd love to get the project team acquainted to your offer and would appreciate if we could schedule an online meeting. Currently, we are pretty busy so we'll have to see when this will be possible, yet let's hope for the best. 
Yours faithfully,
Wolfgang
